pyomo/dae/init_cond.py

```diff
@@ -0,0 +1,174 @@
+                value(condata.lower) - value(condata.body) > tol):
+                inconsistent.add(condata)
+
+    for blk in model.component_objects(Block, active=True):
+        # What if there are time-indexed blocks at multiple levels
+        # of a hierarchy?
+        # My preferred convention is to only check the first (highest-
+        # level) time index, but distinguishing between different-level
+        # time indices is an expensive operation.
+        if not is_explicitly_indexed_by(blk, time):
+            continue
+        info = get_index_set_except(blk, time)
+        non_time_set = info['set_except']
+        index_getter = info['index_getter']
+        for non_time_index in non_time_set:
+            index = index_getter(non_time_index, t0)
+            try:
+                blkdata = blk[index]
+            except KeyError:
+                # Is there some equivalent Block.Skip-like object?
+                if not suppress_warnings:
+                    print(index_warning(blk.name, index))
+                if not allow_skip:
+                    raise
+                continue
+            for condata in blkdata.component_data_objects(Constraint, 
+                    active=True):
+                if (value(condata.body) - value(condata.upper) > tol or
+                    value(condata.lower) - value(condata.body) > tol):
+                    if condata in inconsistent:
+                        raise ValueError(
+                            '%s has already been visited. The only way this '
+                            'should happen is if the model has nested time-'
+                            'indexed blocks, which is not supported.')
+                    inconsistent.add(condata)
+
+    return list(inconsistent)
+
+
+def solve_consistent_initial_conditions(model, time, solver):
+    """
+    """
+    # Need to deactivate discretization equations, wrt time, at t == 0
+    # This is challenging as the only way (to my knowledge) to do this
+    # is to identify_variables in the expression, find the (assume only one?)
+    # DerivativeVar, and access its get_continuousset_list
+    # I would like a get_continuousset_list for discretization equations.
+    # Possibly as a ComponentMap, possibly as an attribute of some new 
+    # DiscEquation subclass of Constraint
+    # Until I have this, this function will only work for backward
+    # discretization schemes
+
+    # Also, would like to be able to check for zero degrees of freedom here
+
+    scheme = time.get_discretization_info()['scheme']
+    if not scheme == 'LAGRANGE-RADAU' or scheme == 'BACKWARD Difference':
+        raise NotImplementedError(
+            '%s discretization scheme is not supported' % scheme)
+
+    t0 = time.first()
+    timelist = [t for t in time if t != t0]
+    was_originally_active = ComponentMap(
+            [(comp, comp.active) for comp in 
+                model.component_data_objects((Block, Constraint))])
```

Comment:
Do you even need this? The `deactivate_model_at` function is only looking for Blocks and Constraints that are currently active so I don't see how anything that was deactivated in the original model would end up in the `deactivated_dict`

Reply:
Good point. This is unnecessary. Removed.